awscli/customizations/s3/filegenerator.py

```diff
@@ -320,7 +323,11 @@ def _list_single_object(self, s3_path):
         # instead use a HeadObject request.
         bucket, key = find_bucket_key(s3_path)
         try:
-            response = self._client.head_object(Bucket=bucket, Key=key)
+            params = {'Bucket': bucket, 'Key': key}
+            if self.sse_c:
```

Comment:
Yeah its really unfortunate. I am not sure about the second option. These reason we do ``HeadObject`` for a single object is that it is much cheaper than ``ListObjects``. We are not necessarily doing it to get metadata. I think I am going to do something along the lines of the first option.